commands/new_function.go

```diff
@@ -103,39 +122,35 @@ the "Dockerfile" lang type in your YAML file.
 	if strings.ToLower(lang) != "dockerfile" {
 		builder.CopyFiles("./template/"+lang+"/function/", "./"+functionName+"/", true)
 	} else {
-		ioutil.WriteFile("./"+functionName+"/Dockerfile", []byte(`FROM alpine:3.6
-# Use any image as your base image, or "scratch"
-# Add fwatchdog binary via https://github.com/openfaas/faas/releases/
-# Then set fprocess to the process you want to invoke per request - i.e. "cat" or "my_binary"
-
-ADD https://github.com/openfaas/faas/releases/download/0.6.9/fwatchdog /usr/bin
-# COPY ./fwatchdog /usr/bin/
-RUN chmod +x /usr/bin/fwatchdog
-
-# Populate example here - i.e. "cat", "sha512sum" or "node index.js"
-ENV fprocess="wc -l"
-
-HEALTHCHECK --interval=5s CMD [ -e /tmp/.lock ] || exit 1
-CMD ["fwatchdog"]
-`), 0600)
+		if err := writeDockerfile(functionName); err != nil {
+			fmt.Printf("Error writing Dockerfile to %s: %v\n", functionName, err)
+		}
 	}
 
-	stack := `provider:
-  name: faas
-  gateway: ` + gateway + `
-
-functions:
+	var stackFileText string
+	functionYAMLData := `
   ` + functionName + `:
     lang: ` + lang + `
     handler: ./` + functionName + `
     image: ` + functionName + `
 `
 
+	if !existingYaml {
+		stackFileText = newStackFileText(functionYAMLData, gateway)
+	} else {
+		existingYAMLText, err := readFileToString(stackFileName)
+		if err != nil {
+			return fmt.Errorf("Could not read stack file: %s\n", stackFileName)
+		}
+
+		stackFileText = insertFunctionIntoYAML(existingYAMLText, functionYAMLData)
+	}
+
 	fmt.Printf(aec.BlueF.Apply(figletStr))
```

Comment:
Kinda of out of scope here, but I don't think we should be printing the logo for commands other than help, since it makes scripting with `faas-cli` messy - @alexellis wdyt?